cognee/api/v1/cognify/code_graph_pipeline.py

```
@@ -0,0 +1,110 @@
+
+async def code_graph_pipeline(datasets: Union[str, list[str]] = None, user: User = None):
+    if user is None:
+        user = await get_default_user()
+
+    existing_datasets = await get_datasets(user.id)
+
+    if datasets is None or len(datasets) == 0:
+        # If no datasets are provided, cognify all existing datasets.
+        datasets = existing_datasets
+
+    if type(datasets[0]) == str:
```

Ruff

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)